### ISSUE TITLE
Add 'decodeEither' function

### ISSUE THREAD

**Issue by chshersh** [haskell/base16-bytestring#6]
This function is convenient. Here is the sample implementation of this function:

```haskell
newtype Decode16Error = Decode16Error ByteString

-- | Decode base16-encoded ByteString.
decodeEither :: ByteString -> Either Decode16Error ByteString
decodeEither = handleError . decode
  where
    handleError (res,rest)
      | BS.null rest = Right res
      | otherwise = Left $ Decode16Error $ "suffix is not in base-16 format: " <> rest
```

**Comment by emilypi** [comment]
@chshersh I'm addressing this in #10. My counterproposal is that the signature for decodes should change entirely to be flush with the other base encoding libraries which provide offsets pointing at the invalid chars, instead of slices of the bytestring.

**Comment by chshersh** [comment]
@emilypi Thanks for working on this!

**Comment by emilypi** [comment]
@chshersh released v1.0.0.0 to hackage 😄
